packages/extension/src/companion/CompanionMenu.tsx

```diff
@@ -270,6 +303,15 @@ export default function CompanionMenu({
           {...parentComment}
         />
       )}
+      {sharePost && (
```

Comment:
Yep, I think at the moment we don't really have any way around that, because it's technically three different apps.
(webapp/extension/companion)

But do agree it would be super nice if they can have their own context wrapper.
Also looping in @sshanzel for this eyes on this.

Reply:
I think that is a great idea to consolidate our modals. Though it may restrict us to perform such specifics as we have had in the past that some properties differ based on where it was implemented. As Chris mentioned, we technically have 3 different apps, we have encountered scenarios where props for the Companion are a bit different from the others (and on some other parts as well).

I really like the idea as it makes things much cleaner in terms of implementation though I think the benefits (flexibility) we have now for providing those values whenever needed currently outweigh the benefits we can have for having a context. Just my two cents about it.

Reply:
Flexibility point is really interesting, I will add context for modals to our improvements ideas page, so we don't lose is

Reply:
Totally agree. After the implementation of our new modal guideline, we can definitely have some discussion around this area. Thank you for adding it to the ideas page.